src/Handler/Table/Preview/PreviewTableHandler.php

```diff
@@ -64,44 +74,32 @@ public function __invoke(
             /** @var string $databaseName */
             $databaseName = $command->getPath()[0];
 
-            // build sql
+            // validate
             $columns = ProtobufHelper::repeatedStringToArray($command->getColumns());
             assert($columns === array_unique($columns), 'PreviewTableCommand.columns has non unique names');
-            $columnsSql = implode(', ', array_map([TeradataQuote::class, 'quoteSingleIdentifier'], $columns));
-
-            $limitSql = sprintf(
-                'TOP %d',
-                ($command->getLimit() > 0 && $command->getLimit() < self::MAX_LIMIT)
-                    ? $command->getLimit()
-                    : self::MAX_LIMIT
-            );
-
-            // TODO changeSince, changeUntil
-            // TODO fulltextSearch
-            // TODO whereFilters
-            // TODO truncated: rewrite to SQL
-            $selectTableSql = sprintf(
-                "SELECT %s %s\nFROM %s.%s",
-                $limitSql,
-                $columnsSql,
-                TeradataQuote::quoteSingleIdentifier($databaseName),
-                TeradataQuote::quoteSingleIdentifier($command->getTableName())
-            );
-
+            if ($command->getLimit() === 0) {
+                $command->setLimit(self::DEFAULT_LIMIT);
+            }
+            if ($command->getLimit() > self::MAX_LIMIT) {
+                $command->setLimit(self::MAX_LIMIT);
```

Comment:
nemelo by to rvat? ostatni backendy to nechavaji byt (je tam stejne check v RO), ale prijde mi to divny chovani

Reply:
jakoze bych tam dal assert a nazdar? Asi muze byt, ale prijde mi to zbytecne striktni..

Reply:
ale OK, ostatni backendy to tak maji:
```php
// \Keboola\Test\Backend\CommonPart1\DataPreviewLimitsTest::testDataPreviewMaximumLimit
            $this->_client->getTableDataPreview($tableId, [
                'limit' => 1200,
            ]);
            $this->fail('limit 1200 should not be allowed');
        } catch (ClientException $e) {
            $this->assertEquals(400, $e->getCode());
            $this->assertEquals('storage.tables.validation', $e->getStringCode());
            $this->assertStringContainsString('1000', $e->getMessage());
        }
```

Reply:
fixnuto